standard/namespace.lua

```diff
@@ -6,18 +6,66 @@
 -- Please see https://github.com/Liquipedia/Lua-Modules to contribute
 --
 
+local Array = require('Module:Array')
 local Class = require('Module:Class')
 local FnUtil = require('Module:FnUtil')
 local String = require('Module:StringUtils')
 local Table = require('Module:Table')
 
+local NS_MAIN = 0
+local NS_USER = 2
+local NS_PROJECT = 4
+local NS_TEMPLATE = 10
+local NS_HELP = 12
+local NS_MODULE = 828
+local NS_MODULE_TALK = 829
+
 local Namespace = {}
 
----Determines if the page this module is invoked on is in main space
----829 (ModuleTalk) is treated as main space for ScribuntoUnit to work properly
+---Determins whether a given title object is in a given namespace, and optionally it's talk namespace.
+---@param title Title
+---@param namespace string|number
+---@param includeTalk boolean?
+---@return boolean
+local function isInNamespace(title, namespace, includeTalk)
+	local fn = includeTalk and title.hasSubjectNamespace or title.inNamespace
+	return fn(title, namespace)
+end
+
+---Determines if a title object is in the Main namespace.
+---`NS_MODULE_TALK` is treated as Main namespace for ScribuntoUnit to work properly.
+---Will use the title object of the page this module is invoked on if no title is provided.
+---@param title Title?
+---@return boolean
+function Namespace.isMain(title)
+	title = title or mw.title.getCurrentTitle()
+	return isInNamespace(title, NS_MAIN) or isInNamespace(title, NS_MODULE_TALK)
```

Comment:
do we even need the "module talk considered as main" hack anymore?
didn't we move the testing into busted etc?